der/src/asn1/real.rs

```diff
@@ -0,0 +1,987 @@
+        // Special value from section 8.5.9 if non zero
+        if self.is_nan()
+            || self.is_infinite()
+            || (self.is_sign_negative() && -self < f64::MIN_POSITIVE)
+            || (self.is_sign_positive() && (*self) < f64::MIN_POSITIVE)
+        {
+            if self.is_sign_positive() && (*self) < f64::MIN_POSITIVE {
+                // Zero
+                return Ok(());
+            } else if self.is_nan() {
+                // Not a number
+                encoder.bytes(&[0b0100_0010])?;
+            } else if self.is_infinite() {
+                if self.is_sign_negative() {
+                    // Negative infinity
+                    encoder.bytes(&[0b0100_0001])?;
+                } else {
+                    // Plus infinity
+                    encoder.bytes(&[0b0100_0000])?;
+                }
+            } else {
+                // Minus zero
+                encoder.bytes(&[0b0100_0011])?;
+            }
+        } else {
+            // Always use binary encoding, set bit 8 to 1
+            let mut first_byte = 0b1000_0000;
+            if self.is_sign_negative() {
+                // Section 8.5.7.1: set bit 7 to 1 if negative
+                first_byte |= 0b0100_0000;
+            }
+            // Bits 6 and 5 are set to 0 to specify that binary encoding is used
+
+            // NOTE: the scaling factor is only used to align the implicit point of the mantissa.
+            // This is unnecessary in DER because the base is 2, and therefore necessarily aligned.
+            // Therefore, we do not modify the mantissa in anyway after this function call, which
+            // already adds the implicit one of the IEEE 754 representation.
+            let (_sign, exponent, mantissa) = decode_f64(*self);
+
+            // Encode the exponent as two's complement on 16 bits and remove the bias
+            let exponent_bytes = exponent.to_be_bytes();
+            let ebytes = strip_leading_zeroes(&exponent_bytes);
+
+            match ebytes.len() {
+                0 | 1 => {}
+                2 => first_byte |= 0b0000_0001,
+                3 => first_byte |= 0b0000_0010,
+                _ => todo!("support multi octet exponent encoding"),
+            }
+
+            encoder.bytes(&[first_byte])?;
+
+            // Encode both bytes or just the last one, handled by encode_bytes directly
+            // Rust already encodes the data as two's complement, so no further processing is needed
+            encoder.bytes(ebytes)?;
+
+            // Now, encode the mantissa as unsigned binary number
+            let mantissa_bytes = mantissa.to_be_bytes();
+            let mbytes = strip_leading_zeroes(&mantissa_bytes);
+            encoder.bytes(mbytes)?;
+        }
+        Ok(())
+    }
+}
+
+impl FixedTag for f64 {
+    const TAG: Tag = Tag::Real;
+}
+
+impl<'a> From<f64> for Any<'a> {
+    fn from(_: f64) -> Any<'a> {
+        Any::from_tag_and_value(Tag::Real, ByteSlice::default())
+    }
+}
```

Comment:
This seems broken? Since `Any` borrows I'm not sure you can define this properly without using `unsafe`, in which case I'd suggest just deleting it.